### PR TITLE
MSI-2190: Configurable product price is always 0 if 'Display Out of Stock Products' = Yes

### PR DESCRIPTION
**Pull request by malesh-v**
### Fixed Issue
#2190 
I've fixed issue with zero price.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/InventoryConfigurableProduct/Model/ResourceModel/Product/LinkedProductStockStatusSelectProcessor.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+use Magento\Catalog\Model\ResourceModel\Product\BaseSelectProcessorInterface;
+use Magento\Framework\DB\Select;
+
+class LinkedProductStockStatusSelectProcessor implements BaseSelectProcessorInterface
+{
+    /**
+     * @param Select $select
+     * @return Select
+     */
+    public function process(Select $select): Select
```
Comment (Stepa4man) [inline app/code/Magento/InventoryConfigurableProduct/Model/ResourceModel/Product/LinkedProductStockStatusSelectProcessor.php:19]: Please add a descriptive comment to this method.

File: app/code/Magento/InventoryConfigurableProduct/Model/ResourceModel/Product/LinkedProductStockStatusSelectProcessor.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfigurableProduct\Model\ResourceModel\Product;
+
+use Magento\Catalog\Model\ResourceModel\Product\BaseSelectProcessorInterface;
+use Magento\Framework\DB\Select;
+
+class LinkedProductStockStatusSelectProcessor implements BaseSelectProcessorInterface
```
Comment (Stepa4man) [inline app/code/Magento/InventoryConfigurableProduct/Model/ResourceModel/Product/LinkedProductStockStatusSelectProcessor.php:13]: Please add annotation block to this class

File: app/code/Magento/InventoryConfigurableProduct/Model/ResourceModel/Product/StockStatusBaseSelectProcessor.php [hunk 3]
```diff
@@ -88,33 +88,31 @@ public function __construct(
      */
```
Comment (Stepa4man) [inline app/code/Magento/InventoryConfigurableProduct/Model/ResourceModel/Product/StockStatusBaseSelectProcessor.php:88]: Please add a descriptive comment to this method.

**Review by Stepa4man — CHANGES_REQUESTED** [review]
Hi @vad1m777, Thanks for your contribution!
Please check my comments

**Comment by sidolov** [comment]
Hi @vad1m777 , a lot of tests are failing with the current solution, any chance to take a look at them? Thank you!

**Comment by sidolov** [comment]
Hi @vad1m777 , due to changes in the repository structure (Magento core code is extracted from the MSI code base) we have changed the base branch to `1.1-develop-legacy`. We will keep this branch for a while to make a smooth transition, in the meanwhile, please, apply changes from the current PR to the new code base structure in `1.1-develop` branch. 

[Installation Guide](https://github.com/magento/inventory/wiki/Metapackage-Installation-Guide) may be helpful for local project development. 

Let me know if you need help or have questions!
Thank you for your collaboration!

**Comment by engcom-Charlie** [comment]
I've closed PR because the issue was closed.

**Pull request closed without merging**